guides/keras_cv/3d_object_detection_with_keras_cv.py

```diff
@@ -0,0 +1,367 @@
+"""
+Title: 3D Object Detection with KerasCV
+Author: Ian Stenbit, Zhaoqi Leng (Waymo), Guowang Li (Waymo)
+Date created: 2023/04/27
+Last modified: 2023/04/27
+Description: Use KerasCV to train a 3D object detection model for LIDAR data.
+Accelerator: GPU
+"""
+
+"""
+KerasCV offers a set of APIs to train LIDAR-based 3D object detection models,
+including dataloading, augmentation, model training, and metric evaluation.
+problems. These APIs were designed and implemented in partnership with Waymo.
+
+In this guide, we'll take KerasCV's 3D object detection API for a spin by training
+a CenterPillar model for Waymo's Open Dataset, which is a 3D object detection
+task for detecting cars and pedestrians for an autonomous vehicle.
+"""
+
+"""shell
+!pip install --upgrade git+https://github.com/keras-team/keras-cv
+!pip install tensorflow==2.11.0
+!pip install waymo-open-dataset-tf-2.11.0==1.5.1
+"""
+
+import keras_cv
+from keras_cv.callbacks import WaymoEvaluationCallback
+
+from keras_cv.datasets.waymo import load
+from keras_cv.datasets.waymo import transformer
+
+from keras_cv.layers.object_detection_3d import voxel_utils
+
+from keras_cv.layers import CenterNetLabelEncoder
+from keras_cv.layers import DynamicVoxelization
+
+from keras_cv.models.__internal__.unet import UNet
+
+from keras_cv.models.object_detection_3d.center_pillar import MultiClassDetectionHead
+from keras_cv.models.object_detection_3d.center_pillar import MultiClassHeatmapDecoder
+from keras_cv.models.object_detection_3d.center_pillar import MultiHeadCenterPillar
+
+import tensorflow as tf
+from tensorflow import keras
+
+"""
+3D object detection is the process of identifying, classifying,
+and localizing objects within a 3D space.  Inputs are often in the form of point
+clouds, although 2D images are sometimes used as inputs as well for multi-model
```

Comment:
perhaps instead of multi-model we say multi-input, or multi-modal, or something else.